proxy/logging/LogField.cc

```diff
@@ -600,23 +596,14 @@ LogField::marshal_agg(char *buf)
 unsigned
 LogField::unmarshal(char **buf, char *dest, int len, LogEscapeType escape_type)
 {
-  if (!m_alias_map) {
-    if (m_unmarshal_func == reinterpret_cast<UnmarshalFunc>(LogAccess::unmarshal_str) ||
-        m_unmarshal_func == reinterpret_cast<UnmarshalFunc>(LogAccess::unmarshal_http_text)) {
-      UnmarshalFuncWithSlice func = reinterpret_cast<UnmarshalFuncWithSlice>(m_unmarshal_func);
-      if (escape_type == LOG_ESCAPE_JSON) {
-        if (m_unmarshal_func == reinterpret_cast<UnmarshalFunc>(LogAccess::unmarshal_str)) {
-          func = reinterpret_cast<UnmarshalFuncWithSlice>(LogAccess::unmarshal_str_json);
-        } else if (m_unmarshal_func == reinterpret_cast<UnmarshalFunc>(LogAccess::unmarshal_http_text)) {
-          func = reinterpret_cast<UnmarshalFuncWithSlice>(LogAccess::unmarshal_http_text_json);
-        }
-      }
-      return (*func)(buf, dest, len, &m_slice);
-    }
-    return (*m_unmarshal_func)(buf, dest, len);
-  } else {
-    return (*m_unmarshal_func_map)(buf, dest, len, m_alias_map);
+  if (std::holds_alternative<UnmarshalFuncWithSlice>(m_unmarshal_func)) {
+    return (*std::get<UnmarshalFuncWithSlice>(m_unmarshal_func))(buf, dest, len, &m_slice);
+  } else if (std::holds_alternative<UnmarshalFuncWithSlice>(m_unmarshal_func)) {
+    return (*std::get<UnmarshalFunc>(m_unmarshal_func))(buf, dest, len);
```

Comment:
This seems to drop the json escape type support.  Also, you could use the variant visit pattern here instead of checking each hold type.

Reply:
Will fix the JSON support.  To me std::visit just makes it harder to read when it's not necessary.

Reply:
JSON support fixed.

Reply:
You can make this a bit simpler using `std::get_if`.
```
if (auto * umf = std::get_if<UnmarshalFuncWithSlice>(m_unmarshal_func) ; umf) {
  return (*umf)(buf, ...);
} else if ( ...)
```